src/cpp/src/image_generation/models/unet_inference_static_bs1.hpp

```diff
@@ -0,0 +1,144 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "lora_helper.hpp"
+#include "image_generation/models/unet_inference.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace genai {
+
+// Static Batch-Size 1 variant of UNetInference
+class UNet2DConditionModel::UNetInferenceStaticBS1 : public UNet2DConditionModel::UNetInference {
+public:
+    virtual void compile(std::shared_ptr<ov::Model> model,
+                         const std::string& device,
+                         const ov::AnyMap& properties) override {
+
+        // All shapes for input/output tensors should be static. 
+        // Double check this and throw runtime error if it's not the case.
+        for (auto& input : model->inputs()) {
+            if (input.get_partial_shape().is_dynamic()) {
+                throw std::runtime_error(
+                    "UNetInferenceStaticBS1::compile: input tensor " + input.get_any_name() +
+                    " shape is dynamic. Tensors must be reshaped to be static before compile is invoked.");
+            }
+        }
+
+        for (auto& output : model->outputs()) {
+            if (output.get_partial_shape().is_dynamic()) {
+                throw std::runtime_error(
+                    "UNetInferenceStaticBS1::compile: output tensor " + output.get_any_name() +
+                    " shape is dynamic. Tensors must be reshaped to be static before compile is invoked.");
+            }
+        }
+
+        // we'll create a separate infer request for each batch.
+        m_nativeBatchSize = model->input("sample").get_shape()[0];
+        m_requests.resize(m_nativeBatchSize);
+
+        //reshape to batch-1
+        UNetInference::reshape(model, 1);
+
+        ov::Core core = utils::singleton_core();
+        ov::CompiledModel compiled_model = core.compile_model(model, device, properties);
+
+        for (int i = 0; i < m_nativeBatchSize; i++ )
+        {
+            m_requests[i] = compiled_model.create_infer_request();
+        }
+    }
+
+    virtual void set_hidden_states(const std::string& tensor_name, ov::Tensor encoder_hidden_states) override {
+        OPENVINO_ASSERT(m_nativeBatchSize && m_nativeBatchSize == m_requests.size(),
+                        "UNet model must be compiled first");
+
+        size_t encoder_hidden_states_bs = encoder_hidden_states.get_shape()[0];
+        if (encoder_hidden_states_bs != m_nativeBatchSize)
```

Comment:
please, use `OPENVINO_ASSERT`

Reply:
addressed in this PR: https://github.com/openvinotoolkit/openvino.genai/pull/1172